internal/inspect/policy.go

```diff
@@ -0,0 +1,315 @@
+// Copyright 2021-2024 Zenauth Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+package inspect
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/google/cel-go/common/ast"
+
+	policyv1 "github.com/cerbos/cerbos/api/genpb/cerbos/policy/v1"
+	responsev1 "github.com/cerbos/cerbos/api/genpb/cerbos/response/v1"
+	runtimev1 "github.com/cerbos/cerbos/api/genpb/cerbos/runtime/v1"
+	"github.com/cerbos/cerbos/internal/compile"
+	"github.com/cerbos/cerbos/internal/namer"
+	"github.com/cerbos/cerbos/internal/policy"
+)
+
+func Policies() *Policy {
+	return &Policy{
+		toResolve: make(map[string]map[string]bool),
+		imports:   make(map[string][]string),
+		results:   make(map[string]*responsev1.InspectPoliciesResponse_Result),
+	}
+}
+
+type Policy struct {
+	toResolve map[string]map[string]bool
+	imports   map[string][]string
+	results   map[string]*responsev1.InspectPoliciesResponse_Result
+}
+
+// Inspect inspects the given policy and caches the inspection related information internally.
+func (pol *Policy) Inspect(p *policyv1.Policy) error {
+	if p == nil {
+		return fmt.Errorf("policy is nil")
+	}
+
+	policyID := namer.PolicyKey(p)
+	variables := policy.ListVariables(p)
```

Comment:
I am not sure if understood what you mean exactly. 

Though I changed the `policy.ListVariables` to return a map instead of a list prevent creating the map afterwards. Because it was only called from a single place and it'd better to return a map for that case.

Reply:
Yeah, it makes sense to get the return value in the shape you need.